staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -157,13 +158,20 @@ var _ http.ResponseWriter = &auditResponseWriter{}
 // create immediately an event (for long running requests).
 type auditResponseWriter struct {
 	http.ResponseWriter
-	event *auditinternal.Event
-	once  sync.Once
-	sink  audit.Sink
+	event      *auditinternal.Event
+	codeOnce   sync.Once
+	headerOnce sync.Once
+	sink       audit.Sink
+}
+
+func (a *auditResponseWriter) setHttpHeader() {
+	a.headerOnce.Do(func() {
+		a.ResponseWriter.Header().Set(auditinternal.HeaderAuditID, string(a.event.AuditID))
```

Comment:
Set [has this behavior already](https://golang.org/pkg/net/http/#Header.Set), but it's not thread-safe, if it's applicable here